packages/cli/src/commands/inspect/index.ts

```diff
@@ -92,20 +99,46 @@ export default async function inspect(client: Client) {
   );
 
   const until = Date.now() + timeout;
-  const wait = argv['--wait'];
+  const wait = parsedArguments.flags['--wait'] ?? false;
+  const withLogs = parsedArguments.flags['--logs'];
 
   // resolve the deployment, since we might have been given an alias
   let deployment = await getDeployment(client, contextName, deploymentIdOrHost);
 
+  let abortController: AbortController | undefined;
+  if (withLogs) {
+    if (wait) {
+      abortController = displayBuildLogs(client, deployment, true);
+    } else {
+      await displayBuildLogs(client, deployment, false);
+      return exitCode(deployment.readyState, print);
```

Comment:
Nit: I saw these `return exitCode(...)` spread in a few places. It would be simpler to read the code if they were all in a single place later in the code that does something like:

```
// only log the status when `--logs` is passed
if (withLogs) {
  // print status
  return exitCode(deployment.readyState)
}

// fetch and print the whole details
```

Reply:
Yes, but I also didn't meant to reshape the whole function. However you're right and I'll make bigger changes.